File: array.cpp
//https://practice.geeksforgeeks.org/problems/minimum-swaps-required-to-bring-all-elements-less-than-or-equal-to-k-together4847/1#

// C++ program to find minimum swaps required
// to club all elements less than or equals
// to k together
#include <bits/stdc++.h>
using namespace std;



 // } Driver Code Ends


class Solution
{
public:
    int minSwap(int arr[], int n, int k) {
        // Complete the function
        int total = 0; // no. of elements <= K
        int swap_count = 0;
        
        for(int i=0; i<n; i++)
        {
            if(arr[i] <= k)
              total++;
        }
        
        // first sliding window
        for(int i=0; i<total; i++)
        {
            if(arr[i] > k)
              swap_count++;
        }
        
        //leaving one element from left and 
        //adding one element in the right part like a sliding window.
        int i = 0;
        int j = total;
        int min_swaps = swap_count; // "min" holds min no. of swaps.
        while(j<n)
        {
            if(arr[i] > k)
              swap_count--;
            
            if(arr[j] > k)
              swap_count++;
            
            min_swaps = min(min_swaps,swap_count);
            
            i++;
            j++;
        }
        return min_swaps;
    }
};


// { Driver Code Starts.

// Driver code
int main() {

	int t,n,k;
	cin>>t;
	while(t--)
    {
        cin>>n;
        int arr[n];
        for(int i=0;i<n;i++)
            cin>>arr[i];
        cin>>k;
        Solution ob;
        cout << ob.minSwap(arr, n, k) << "\n";
    }
	return 0;
}
  // } Driver Code Ends

File: array.py
#https://practice.geeksforgeeks.org/problems/trapping-rain-water-1587115621/1#

        #Code here
        # initialize output
        result = 0
     
        # maximum element on left and right
        left_max = 0
        right_max = 0
     
        # indices to traverse the array
        lo = 0
        hi = n-1
     
        while(lo <= hi): 
            if(arr[lo] < arr[hi]):
                if(arr[lo] > left_max):
                    # update max in left
                    left_max = arr[lo]
                else:
                    # water on curr element = max - curr
                    result += left_max - arr[lo]
                lo+= 1
        
            else:
                if(arr[hi] > right_max):
                    # update right maximum
                    right_max = arr[hi]
                else:
                    result += right_max - arr[hi]
                hi-= 1
        
        return result

#{ 
#  Driver Code Starts
#Initial Template for Python 3

import math



def main():
        T=int(input())
        while(T>0):
            
            n=int(input())
            
            arr=[int(x) for x in input().strip().split()]
            obj = Solution()
            print(obj.trappingWater(arr,n))
            
            
            T-=1


if __name__ == "__main__":
    main()



# } Driver Code Ends